static/app/components/events/errors.tsx

```diff
@@ -174,14 +175,22 @@ class Errors extends Component<Props, State> {
                     return false;
                   });
 
-                  if (releaseArtifact && !!releaseArtifact.dist) {
+                  const releaseArtifactDistribution = releaseArtifact?.dist ?? null;
+
+                  // Neither event nor file have dist -> matching
+                  // Event has dist, file doesn’t -> not matching
+                  // File has dist, event doesn’t -> not matching
+                  // Both have dist, same value -> matching
+                  // Both have dist, different values -> not matching
+                  if (releaseArtifactDistribution !== eventDistribution) {
                     error.message = t(
                       'Source code was not found because the distribution did not match'
                     );
                     data['expected-distribution'] = dist;
```

Comment:
Could we use `eventDistribution` here instead? It just makes it clearer to the reader of the code which is which.

Reply:
According to the typescript, the `dist` of an event can be of type `string` or `null`, but just in case it is `undefined`, I thought it would be better to display the real value here